### PR TITLE
removed more lodash stuff

### PR DESCRIPTION
**Pull request by jimmywarting**
### Description
got rid of more lodash stuff

### Motivation and Context
part of #2187 

### How Has This Been Tested?

- npm run test
- npm run lint

### Types of changes

- [ ] No code changes (changes to documentation, CI, metadata, etc)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Refactor

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/helpers.js [hunk 1]
```diff
@@ -11,7 +10,7 @@ export function isOAS3(spec) {
     return false;
   }
 
-  return startsWith(oasVersion, '3');
+  return oasVersion[0] === '3';
```
Comment (char0n) [inline src/helpers.js:13]: wouldn't following form be more readable in this case?

```js
return oasVersion.charAt(0) === '3';
```

By looking at above `charAt` I guess I'd still prefer a `startsWith` method here:

```js
return oasVersion.startsWith('3')
```
It's declarative and it does exactly what it says. What's your take on this?

File: src/helpers.js [hunk 2]
```diff
@@ -20,7 +19,7 @@ export function isSwagger2(spec) {
     return false;
   }
 
-  return startsWith(swaggerVersion, '2');
+  return swaggerVersion[0] === '2';
```
Comment (char0n) [inline src/helpers.js:22]: Duplicate of https://github.com/swagger-api/swagger-js/pull/2242/files#r709543609

File: src/http/index.js [hunk 3]
```diff
@@ -142,7 +141,7 @@ function serializeHeaderValue(value) {
 //     Cookie: two
 //  =  { Cookie: [ "one", "two" ]
 export function serializeHeaders(headers = {}) {
-  if (!isFunction(headers.entries)) return {};
+  if (typeof headers.entries !== 'function') return {};
```
Comment (char0n) [inline src/http/index.js:144]: I had a brief note on the predicates like this in my original comment: https://github.com/swagger-api/swagger-js/issues/2187

Given that we have `valid-typeof` eslint rule active, the correctness of these assertions is guaranteed; so let's go for it.

File: src/helpers.js (hunk 1 above)
Comment (jimmywarting) [reply src/helpers.js:13]: They all solves the issue. I Just took the most easiest, low level, fastest, and shortest method available. 

I don't really care about which of them are being used.
charAt has better support then newer startswith

My method could work for both string and array of it where splited

Comment (char0n) [reply src/helpers.js:13]: This library minimum required versions of Node.js is now at 12.4. So as long as we use features that falls withing this minimum required version we're fine. 

> I don't really care about which of them are being used.

If you don't have preference let use `startsWith`. It increases the readability a bit.

Comment (jimmywarting) [reply src/helpers.js]: fixed

**Review by char0n — APPROVED** [review]
LGTM. Thank you for contributing!

**Pull request merged**